submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,112 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammals{
+   constructor(name,gender,friends){
+      this.species = 'mammals',
+      this.name = name,
+      this.gender = gender,
+      this.legs = 4,
+      this.friends = friends
+   }
+}
+
+class Dog extends Mammals{
+   constructor(name,gender,friends,saying){
+      super(name,gender,friends);
+      this.species = 'dog',
+      this.saying = saying || 'guf'
+   }
+};
+
+class Cat extends Mammals{
+   constructor(name,gender,friends,saying){
+      super(name,gender,saying,friends);
+      this.species = 'cat',
+      this.saying = saying || 'guf'
+   }
+
+   getSaying(){
+      return this.saying || 'meow';
+   }
+}
+
+class Human extends Mammals{
+   constructor(name,gender,friends,saying){
+      super(name,gender,friends);
+      this.species = 'human'
+      this.hands = 2;
+      this.legs =2;
+      this.saying = saying || `Hello! My name is ${this.name}`
+   }
+}
+
+class Woman extends Human{
+   constructor(name,friends,saying){
+      super(name,friends,saying);
+      this.gender = 'female'
+   }
+}
+
+class Man extends Human{
+   constructor(name,friends,saying){
+      super(name,friends,saying);
+      this.gender = 'male'
+   }
+}
+
+class CatWoman extends Woman{
+   constructor(name,friends,saying){
+      super(name,friends,saying);
+      this.saying = Cat.prototype.getSaying();
+   }
+}
+
+const cat = new Cat(name='Murzic',gender='male',friends=['Elza','Nicky'])
+const dog = new Dog(name='Lucy',gender='female',saying='guffyy',friends=['Alla','Dima']);
+const man = new Man(name='Vlad',friends=['Sergay','Anton'],saying='Hello everyone!');
+const woman = new Woman(name='Anna',friends=['Sergay','Artem']);
+const catWoman = new CatWoman(name='Lily',friends=['Batman'])
```

Comment:
Having no semicolon is generally valid in JS. Yet we recommend to keep code style consitent.

Please take every code review comment and check your entire code against the comment.

Code is written for other people. Please take care of them. Consistent code style is a form of caring.